docs/root/configuration/listeners/network_filters/local_rate_limit_filter.rst

```diff
@@ -0,0 +1,38 @@
+.. _config_network_filters_local_rate_limit:
+
+Local rate limit
+================
+
+* Local rate limiting :ref:`architecture overview <arch_overview_local_rate_limit>`
+* :ref:`v2 API reference
+  <envoy_api_msg_config.filter.network.local_rate_limit.v2alpha.LocalRateLimit>`
+* This filter should be configured with the name *envoy.filters.network.local_ratelimit*.
+
+Overview
+--------
+
+The local rate limit filter applies a :ref:`token bucket
+<envoy_api_field_config.filter.network.local_rate_limit.v2alpha.LocalRateLimit.token_bucket>` rate
+limit to incoming connections. Each connection utilizes a single token, and if no tokens are
+available, the connection will be immediately closed without further filter iteration.
+
+.. _config_network_filters_local_rate_limit_stats:
+
+Statistics
+----------
+
+Every configured local rate limit filter has statistics rooted at *local_ratelimit.<stat_prefix>.*
+with the following statistics:
+
+.. csv-table::
+  :header: Name, Type, Description
+  :widths: 1, 1, 2
+
+  rate_limit, Counter, Total connections that have been closed due to rate limit exceeded
```

Comment:
I thought about this. My concern was that in typical deployments this gauge will only be read very infrequently during stats flush intervals, so I'm not sure how useful it would actually be in practice. I'm happy to add it though. If we did add it, we could have a deterministic integration test also. WDYT?

Reply:
Would this really be useful if the refills are occurring ~50ms? The gauge would be all over the place and we'd have to sample the gauge at least twice every 50ms to avoid aliasing. I can't imagine we'd be able to draw any conclusions at high refill frequencies.

Reply:
Yeah, that was my hesitance. The likely resolution problems in actual deployments make the potential use a little murky. While this might not be particularly useful for Lyft's high refill needs it might provide observability for others. Up to you. If you include we can make a note with these caveats.

Reply:
Yeah I think I'm going to skip this for now.